### PR TITLE
Change PostGenerationMethodCall example

### PR DESCRIPTION
**Pull request by francoisfreitag**
User passwords don’t depend on the generated object. The previous example was tied to Django behavior, while `factory_boy` is a general purpose library. Besides, this example gets in the way of removing the duplicate call to `.save()` after `postgeneration` hooks.

Refs #366

### CODE REVIEW

**Comment by federicobond** [comment]
While I agree with the motivation for this change, I find the Point example a bit contrived. At first I thought `_distance_from_origin` was some kind of cached value, but it's only accessed through the private property, which is unidiomatic. Perhaps a bit more explanation about what you are trying to achieve in the example would be helpful.

**Comment by francoisfreitag** [comment]
I changed the example to be shorter and better illustrate the purpose of `PostGenerationMethodCall`.
The new `DefaultRegistry()` and `OtherRegistry()` classes are not documented, hopefully the example is easy enough to understand and their implementation does not matter.

File: docs/reference.rst [hunk 1]
```diff
@@ -1878,71 +1878,37 @@ Once the factory instance has been generated, the method specified in
 with any arguments specified in the :class:`PostGenerationMethodCall` declaration, by
 default.
 
-For example, to set a default password on a generated User instance
-during instantiation, we could make a declaration for a ``password``
-attribute like below:
+For example, using ``PostGenerationMethodCall`` to register created users in an
+external system.
```
Comment (jdufresne) [inline docs/reference.rst:1882]: I think adding an explicit subject makes this slightly more clear, up to you.

```suggestion
For example, we could use ``PostGenerationMethodCall`` to register created users in an
external system.
```

**Review by jdufresne — APPROVED** [review]
I like the direction #623 is going and this helps us get there.

**Comment by WhyNotHugo** [comment]
I have some usages that were pretty close to the example:

   password = PostGenerationMethodCall("set_password", "123")

As of a recent release, I'm seeing warnings that this behaviour is deprecated, but it's not clear to me how one should implement the previous behaviour -- the new example is completely different,

Any clues as to what I should be using for such scenarios now?

**Comment by WhyNotHugo** [comment]
Ah, looks like this does it:

```py

    @post_generation
    def password(obj, create, extracted, **kwargs):
        obj.set_password("123")
        if create:
            obj.save()
```

**Comment by francoisfreitag** [comment]
https://factoryboy.readthedocs.io/en/latest/orms.html#factory.django.Password is the recommended way.

While using a `post_generation` hook works, it will actually issue two queries:
1. `INSERT users_user VALUES (..., ...)`
2. `UPDATE users_user SET ...`

The second query is not necessary, the password just needs hashing, which django can do with the [`make_password`](https://docs.djangoproject.com/en/dev/topics/auth/passwords/#django.contrib.auth.hashers.make_password).

The warning you’re seeing comes from the deprecation announced in https://factoryboy.readthedocs.io/en/stable/changelog.html#id1. Refs #316.

**Comment by WhyNotHugo** [comment]
`factory.django.Password` is perfect, thanks for the hint!

**Pull request merged**